### PR TITLE
minor provider updates to resolve klocworks issues

### PR DESCRIPTION
**Pull request by shefty**
Mostly adds bound checks, and editor automatically removed some unneeded white space when saving.

### CODE REVIEW

**Comment by rajachan** [comment]
bot:aws:retest

**Comment by shefty** [comment]
@aingerson @j-xiong @ooststep -- please review changes to providers.

File: prov/util/src/util_shm.c [hunk 1]
```diff
@@ -153,13 +153,14 @@ int smr_create(const struct fi_provider *prov, struct smr_map *map,
 	fd = shm_open(attr->name, O_RDWR | O_CREAT, S_IRUSR | S_IWUSR);
 	if (fd < 0) {
 		FI_WARN(prov, FI_LOG_EP_CTRL, "shm_open error\n");
-		goto err1;
+		return -errno;
```
Comment (j-xiong) [inline prov/util/src/util_shm.c:156]: According to 'man errno', the value of errno may change after FI_WARM() is called, even though a successful call doesn't affect errno in my test. Same to other places in this commit.

Comment (shefty) [reply prov/util/src/util_shm.c:156]: This calling sequence is pervasive throughout the code.  We recently applied a patch to fi_log() directly to address this.  fi_log now leaves errno unaffected, by saving and restoring it.

**Comment by shefty** [comment]
bot:aws:retest

**Comment by shefty** [comment]
bot:aws:retest

**Comment by shefty** [comment]
@rajachan - Is there a problem with the AWS CI?  The failure is persistent across retests.

**Pull request merged**